fw/http2.c

```diff
@@ -204,14 +204,110 @@ tfw_h2_cleanup(void)
 	tfw_h2_stream_cache_destroy();
 }
 
+static inline void
+tfw_h2_context_init_sched_enry_storage(TfwH2Ctx *ctx)
+{
+	unsigned int def_sched_entry_count =
+		(2 * PAGE_SIZE - sizeof(TfwH2Ctx)) / sizeof(TfwStreamSchedEntry);
+	void *base = (void *)((char *)ctx + sizeof(TfwH2Ctx));
+
+	/*
+	 * We allocate 2* PAGE_SIZE bytes for each TfwH2Ctx, and
+	 * use free memory after TfwH2Ctx structure for sched
+	 * entries.
+	 */
+	tfw_h2_stream_sched_init_entry_storage(&ctx->sched, base,
+					       def_sched_entry_count);
+}
+
+TfwH2Ctx *
+tfw_h2_context_alloc(void)
+{
+	struct page *pg;
+
+	pg = alloc_pages(GFP_ATOMIC, 1);
```

Comment:
Is this not too much just for H2 context?

I've a doubts about page allocation at this place. I would prefer to use `TfwPool` or memcache.